src/vm/methodtable.cpp

```diff
@@ -3092,8 +3103,14 @@ void  MethodTable::AssignClassifiedEightByteTypes(SystemVStructRegisterPassingHe
                 currentEightByte++;
                 _ASSERTE(currentEightByte <= CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
-                currentEightByteOffset = offset + fieldSize;
-                accumulatedSizeForEightByte = 0;
+                currentEightByteOffset = offset + Min(fieldSize, (unsigned int)SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES);
```

Comment:
I cannot get my head around this change - I think we need to make sure that the `currentEightByteOffset` is set to multiple of `SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES` here (since we support only two eightbytes, it should result in it being set to `SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES`). 
It seems it should be:
```c++
currentEightByteOffset += SYSTEMV_EIGHT_BYTE_SIZE_IN_BYTES;
```